### DISCUSSION TITLE
Extract file upon uninstall

### DISCUSSION THREAD

**Discussion by vincent-thx** [wixtoolset/issues#8136] [Questions] [answered]
Is it possible to have WiX extract a file from the msi payload upon uninstall and then use in a custom action?

**Comment by chrpai** [comment]
What do you plan on writing the custom action in?

**Reply by vincent-thx** [reply]
My custom action dll is written in c#.

**Comment by chrpai** [comment]
For C#/DTF the easiest way I know to do it is just include the file in the C# .csproj as content copy local.  This will package it up inside the CA and place it in the current directory (or subdirectory) and made available when the C# code runs.

**Reply by vincent-thx** [reply]
I already have a working WiX installer with custom action dll. The installer payload contains a zip file that includes the nesseccary files to install an audio driver. The zip file gets extracted to my defined temp folder upon install. My question is if it's possible to extract the zip during uninstall so that I can get at driver inf files that were used during the install. I know I could stash away those files during install for later use but would rather just pull them from the installer exe during uninstall.

**Comment by chrpai** [comment, marked as answer]
I'm saying if you add the contents of your zip to your ca .csproj it'll just be there without any additional work.  If you want to be able to get it from the MSI then you'll have to call a bunch of APIs to locate the MSI (which could be missing),   query a few tables, stream out the cab, extract the zip file from the cab and then extract the contents of the zip.

**Reply by vincent-thx** [reply]
I see. I have this working and it meets our needs. Thanks
